:warning: Stop serving v1alpha3 API types (#8549)
kubernetes-sigs/cluster-api: docs/book/src/developer/providers/migrations/v1.4-to-v1.5.md

## Changes by Kind

### Deprecation
- 

> - API version`v1alpha4` is deprecated and will be removed in v1.6.

Reviewer: ```suggestion
- API version`v1alpha4` is deprecated and CAPI will stop serving this version in v1.6.
```

### Removals

> - API version `v1alpha3` is removed from v1.5.

Reviewer: ```suggestion
- API version `v1alpha3` is not served anymore from v1.5 (users can enable it manually in case they are lagging behind with deprecation cycles). Important: `v1alpha3` will be removed in 1.6.
```

### Deprecation

> - API version `v1alpha3` is not served anymore from v1.5 (users can enable it manually in case they are lagging behind with deprecation cycles). Important: `v1alpha3` will be removed in 1.6.

Reviewer 2: Not sure I understand why v1alpha3 is under deprecation and v1alpha4 under removal

Reviewer 2: That would have been my general feeling :D
